File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/customScripts/CustomBottomSheet.java
package com.atomykcoder.atomykplay.customScripts;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.coordinatorlayout.widget.CoordinatorLayout;

import com.google.android.material.bottomsheet.BottomSheetBehavior;

public class CustomBottomSheet<V extends View> extends BottomSheetBehavior<V> {

    private boolean enableCollapse = false;

    public CustomBottomSheet() {
    }

    public CustomBottomSheet(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public void isEnableCollapse(boolean enableCollapse) {
        this.enableCollapse = enableCollapse;
    }

    @Override
    public boolean onInterceptTouchEvent(@NonNull CoordinatorLayout parent, @NonNull V child, @NonNull MotionEvent event) {
        if (enableCollapse) {
            return false;
        }
        return super.onInterceptTouchEvent(parent, child, event);
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/scripts/CustomBottomSheet.kt
package com.atomykcoder.atomykplay.scripts

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.google.android.material.bottomsheet.BottomSheetBehavior

class CustomBottomSheet<V : View> : BottomSheetBehavior<V> {
    private var enableCollapse = false

    constructor()
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    fun isEnableCollapse(enableCollapse: Boolean) {
        this.enableCollapse = enableCollapse
    }

    override fun onInterceptTouchEvent(
        parent: CoordinatorLayout, child: V, event: MotionEvent
    ): Boolean {
        return if (enableCollapse) {
            false
        } else super.onInterceptTouchEvent(parent, child, event)
    }
}
